alicloud/resource_alicloud_cen_bandwidth_package.go

```diff
@@ -0,0 +1,308 @@
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCenBandwidthPackageCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	var cenbwp *cbn.CreateCenBandwidthPackageResponse
+	err := resource.Retry(3*time.Minute, func() *resource.RetryError {
+		args := buildAliCloudCenBandwidthPackageArgs(d, meta)
+		resp, err := client.cenconn.CreateCenBandwidthPackage(args)
+		if err != nil {
+			if IsExceptedError(err, OperationBlocking) {
+				return resource.RetryableError(fmt.Errorf("Create CEN bandwidth package timeout and got an error: %#v.", err))
+			}
+			return resource.NonRetryableError(err)
+		}
+		cenbwp = resp
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("Create CEN bandwidth package got an error :%#v", err)
+	}
+
+	d.SetId(cenbwp.CenBandwidthPackageId)
+
+	err = client.WaitForCenBandwidthPackage(d.Id(), Idle, DefaultCenTimeout)
+	if err != nil {
+		return fmt.Errorf("Timeout when WaitForCenBandwidthPackageIdle, bandwidth package ID %s", err)
+	}
+
+	return resourceAlicloudCenBandwidthPackageRead(d, meta)
+}
+
+func resourceAlicloudCenBandwidthPackageRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AliyunClient)
+	resp, err := client.DescribeCenBandwidthPackage(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	geographicRegionId := make([]string, 0)
+	geographicRegionId = append(geographicRegionId, convertGeographicRegionId(resp.GeographicRegionAId))
+	geographicRegionId = append(geographicRegionId, convertGeographicRegionId(resp.GeographicRegionBId))
+
+	d.Set("geographic_region_ids", geographicRegionId)
+	d.Set("bandwidth", resp.Bandwidth)
+	d.Set("name", resp.Name)
+	d.Set("description", resp.Description)
+	d.Set("expired_time", resp.ExpiredTime)
+	d.Set("status", resp.Status)
+
+	if resp.BandwidthPackageChargeType == "POSTPAY" {
+		d.Set("charge_type", "PostPaid")
+	} else {
+		d.Set("charge_type", "PrePaid")
+	}
+
+	return nil
+}
+
+func resourceAlicloudCenBandwidthPackageUpdate(d *schema.ResourceData, meta interface{}) error {
+	d.Partial(true)
+
+	client := meta.(*AliyunClient)
+	attributeUpdate := false
+	request1 := cbn.CreateModifyCenBandwidthPackageAttributeRequest()
+	request1.CenBandwidthPackageId = d.Id()
+
+	if d.HasChange("name") {
+		d.SetPartial("name")
+		request1.Name = d.Get("name").(string)
+
+		attributeUpdate = true
+	}
+
+	if d.HasChange("description") {
+		d.SetPartial("description")
+		request1.Description = d.Get("description").(string)
+
+		attributeUpdate = true
+	}
+
+	if attributeUpdate {
+		if _, err := client.cenconn.ModifyCenBandwidthPackageAttribute(request1); err != nil {
+			return err
+		}
+	}
+
+	request2 := cbn.CreateModifyCenBandwidthPackageSpecRequest()
+	request2.CenBandwidthPackageId = d.Id()
+
+	if d.HasChange("bandwidth") {
+		d.SetPartial("bandwidth")
+		bandwidth := d.Get("bandwidth").(int)
+		request2.Bandwidth = requests.NewInteger(bandwidth)
+
```

Comment:
Please move line 204-205 into line 207 block and merge line 209 and 210 into one line.